python/packages/ai/teams/ai/promptsv2/prompt.py

```diff
@@ -0,0 +1,36 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from typing import List
+
+from .layout_engine import LayoutEngine
+from .prompt_section import PromptSection
+
+
+class Prompt(LayoutEngine):
+    """
+    Top level prompt section.
+
+    Prompts are compositional such that they can be nested to create complex prompt hierarchies.
+    """
+
+    def __init__(
+        self,
+        sections: List[PromptSection],
+        tokens: float = -1,
```

Comment:
Not too familiar with this, but would it be good practice to do Optional[x] if x is optional for subclasses as well?

Reply:
Do you mean change to `tokens: Optional[float] = None`? Since we need to have default values for this parameter, the current implementation should be cleaner and shorter. And users can also see the default value without going through our code.

Changing to `Optional[float]` also does not have improvement on the actual coding experience, users can skip this parameter no matter whether we mark it as Optional or not. And subclass can also skip this parameter if it wants to use the default value and does not allow users to change it.

Please let me know if you're referring to other ideas.